utils/s3/client.hh

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <seastar/core/sstring.hh>
+#include <seastar/core/shared_ptr.hh>
+#include <seastar/http/client.hh>
+
+using namespace seastar;
+
+namespace s3 {
+
+struct range {
+    uint64_t off;
+    size_t len;
+};
+
+class client : public enable_shared_from_this<client> {
+    sstring _host;
+    http::experimental::client _http;
+
+    struct private_tag {};
+    future<std::vector<temporary_buffer<char>>> get_object(sstring object_name, std::optional<std::pair<uint64_t, size_t>> range);
+
+public:
+    explicit client(ipv4_addr addr, private_tag);
+    static shared_ptr<client> make(ipv4_addr addr);
+
+    future<uint64_t> get_object_size(sstring object_name);
+    future<temporary_buffer<char>> get_object_contiguous(sstring object_name, std::optional<range> range = {});
+    future<> put_object(sstring object_name, temporary_buffer<char> buf);
+    future<> delete_object(sstring object_name);
+
```

Comment:
Will we need an abort() here? Usually disk I/O is expected to complete in finite time, but network partition can cause us to time out here. On the other hand, if we're partitioned, nobody cares if we hang during shutdown or not.

I think we can avoid adding abort(), and if it turns out to be a problem, add tcp keepalive.